### PR TITLE
Added specs for Gem

### PR DESCRIPTION
**Pull request by mishuagopian**
## Summary
- Added specs for gem

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 AllCops:
   Exclude:
     - wor-requests.gemspec
+    - spec/**/*
```
Comment (alebian) [inline .rubocop.yml]: :unamused: :smirk:

File: spec/spy.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#   Spy.spy(controller, 'authenticate_entity')
+#   perform_some_request
+# end
+#
+# it 'calls authenticate_entity defined by the user' do
+#   count = controller.spied_authenticate_entity_counter
+#   expect(count).to be(1)
+# end
+#
+class Spy
```
Comment (alebian) [inline spec/spy.rb]: emoushon

File: spec/wor/requests/base_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# require 'spec_helper'
```
Comment (alebian) [inline spec/wor/requests/base_spec.rb]: Why is this file all commented out? Delete it if it is not necessary

File: spec/wor/test_services/some_service.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    e
+  end
+
+  def get_without_rescue
+    get(
+      path: '/mypath'
+    )
+  end
+
+  VALID_HTTP_VERBS = %i(get post patch put delete).freeze
```
Comment (alebian) [inline spec/wor/test_services/some_service.rb]: You can use the array defined in Wor::Requests::Base::VALID_HTTP_VERBS. I think this line is not necessary

**Review by alebian — CHANGES_REQUESTED** [review]

File: spec/wor/test_services/some_service_failure_spec.rb [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            Spy.spy(service.logger, 'info')
+            Spy.spy(service.logger, 'error')
+            @response = service.get_without_log_with_block
+          end
+
+          it 'raises no exception' do
+            expect{ @response }.not_to raise_error
+          end
+
+          it 'logger.info has been called once' do
```
Comment (Enanodr) [inline spec/wor/test_services/some_service_failure_spec.rb]: if logger.info has been called once why are you checking if eq(0) ?

File: spec/wor/test_services/some_service_failure_spec.rb [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            expect(service.logger.spied_info_counter).to be(0)
+          end
+
+          it 'logger.error has been called once' do
+            expect(service.logger.spied_error_counter).to be(0)
```
Comment (Enanodr) [inline spec/wor/test_services/some_service_failure_spec.rb]: idem

File: spec/wor/test_services/some_service_successful_spec.rb [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            Spy.spy(service.logger, 'info')
+            @response = service.get_without_log_with_block
+          end
+
+          it 'raises no exception' do
+            expect{ @response }.not_to raise_error
+          end
+
+          it 'logger.info has been called twice' do
+            expect(service.logger.spied_info_counter).to be(0)
```
Comment (Enanodr) [inline spec/wor/test_services/some_service_successful_spec.rb]: idem

**Review by draffo-dc — CHANGES_REQUESTED** [review]

File: .rubocop.yml (hunk 1 above)
Comment (mishuagopian) [inline .rubocop.yml]: 👼

**Comment by mishuagopian** [comment]
@Enanodr @alebian back to you guys

**Review by alebian — APPROVED** [review]
Looks good to me! :+1:

**Review by draffo-dc — APPROVED** [review]

**Pull request merged**